src/components/Post/GovernanceSideBar/Referenda/VoteReferendum.tsx

```diff
@@ -246,23 +280,81 @@ const VoteReferendum = ({ className, referendumId, onAccountChange, lastVote, se
+
+			if(vote === EVoteDecisionType.AYE ) {
+				try {
+					voteTx = api.tx.convictionVoting.vote(referendumId, { Standard: { balance: lockedBalance, vote: { aye:true, conviction } } });
+				} catch (e) {
+					console.log(e);
+				}
+			}
+			else if(vote === EVoteDecisionType.NAY ) {
+				try {
+					voteTx = api.tx.convictionVoting.vote(referendumId, { Standard: { balance: lockedBalance, vote: { aye:false, conviction } } });
+				} catch (e) {
+					console.log(e);
+				}
+			}
+
+			else if(vote === EVoteDecisionType.SPLIT) {
+				try {
+					await splitForm.validateFields();
+
+					// if form is valid
+					const  ayeVote = ayeVoteValue?.toString();
+					const  nayVote = nayVoteValue?.toString();
+					voteTx = api.tx.convictionVoting.vote(referendumId, { Split: { aye:`${ayeVote}`,nay:`${nayVote}` } });
+				} catch (e) {
+					console.log(e);
+				}
+				finally{
+					setAyeVoteValue(undefined);
+					setNayVoteValue(undefined);
+				}
+			}
+
+			else if(vote === EVoteDecisionType.ABSTAIN && ayeVoteValue && nayVoteValue) {
+				try {
+					await splitForm.validateFields();
+					// if form is valid
+					const  abstainVote = abstainVoteValue?.toString();
+					const  ayeVote = ayeVoteValue?.toString();
+					const  nayVote = nayVoteValue?.toString();
+					voteTx = api.tx.convictionVoting.vote(referendumId, { SplitAbstain: {  abstain:`${abstainVote}`,aye:`${ayeVote}`, nay:`${nayVote}` } });
+				} catch (e) {
+					console.log(e);
+				}
+				finally{
+					setAbstainVoteValue(undefined);
+					setNayVoteValue(undefined);
+					setAyeVoteValue(undefined);
+				}
+			}
 		} else if(proposalType === ProposalType.FELLOWSHIP_REFERENDUMS) {
-			voteTx = api.tx.fellowshipCollective.vote(referendumId, aye);
-		} else{
-			voteTx = api.tx.democracy.vote(referendumId, { Standard: { balance: lockedBalance, vote: { aye, conviction } } });
+			if(vote === EVoteDecisionType.AYE){
+				voteTx = api.tx.fellowshipCollective.vote(referendumId, true);
+			}else{
+				voteTx = api.tx.fellowshipCollective.vote(referendumId, false);
+			}
+		}
+		else{
+			if(vote === EVoteDecisionType.AYE){
+				voteTx = api.tx.democracy.vote(referendumId, { Standard: { balance: lockedBalance, vote: { aye:true, conviction } } });
+			}
+			else{
+				voteTx = api.tx.democracy.vote(referendumId, { Standard: { balance: lockedBalance, vote: { aye:false , conviction } } });
+			}
 		}
-
 		if(network == 'equilibrium'){
-			voteTx.signAndSend(address, { nonce: -1 }, ({ status }) => {
+			voteTx?.signAndSend(address, { nonce: -1 }, ({ status }) => {
 				if (status.isInBlock) {
 					setLoadingStatus({ isLoading: false, message: '' });
 					queueNotification({
 						header: 'Success!',
 						message: `Vote on referendum #${referendumId} successful.`,
 						status: NotificationStatus.SUCCESS
 					});
-					setLastVote(aye ? 'aye' : 'nay');
+					setLastVote(vote);
```

Comment:
setLastVote compatibility check for parent component

Reply:
Yes sir